### DISCUSSION TITLE
Solana program integration using wormhole.

### DISCUSSION THREAD

**Discussion by aravinddojima** [wormhole-foundation/wormhole#3922] [Developer Q&A]
## Description and context
I hardly able to see the flow of solana wormhole transactions.
If I want to integrate my solana program with wormhole, What is the process ?
Are there any docs for solana integration ?
What's happening inside solana modules when we integrate our programs ?

Can anyone answer the above questions ?

EVM part was clear but I don't find enough docs on solana integration.

**Comment by evan-gray** [comment]
Perhaps take a look at [Wormhole Scaffolding](https://github.com/wormhole-foundation/wormhole-scaffolding) for getting started on Solana.

**Reply by aravinddojima** [reply]
But it doesn't clearly say what's happening behind the scenes. 
Will relayer be calling the receive_message function on solana (or) core contracts are calling the receive_message function ?
Can I get some documents depicting the flow of the programs in solana and sui ?
@evan-gray

**Reply by evan-gray** [reply]
I have raised with some of the contributors / maintainers that it would be good to have docs to outline these flows. 

On almost every chain, a relayer (be it a Standard Relayer, an off-chain process, or an end user via a web app) would be responsible for calling your contract. 

To my understanding…

On Solana, the VAA must first be verified via the core bridge - this typically involves several instructions over a few transactions to verify the signatures and post the, now verified, VAA (which is stored to a PDA of the core bridge). Your application would then read that account (verifying its derivation) by providing it as one of the accounts in an instruction. The most modern, open-source Solana code I know of to integrate directly with the core bridge was [NTT](https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/solana/programs/example-native-token-transfers/src/transceivers/wormhole/instructions/receive_message.rs#L39)

On Sui, the flow is similar, in that a transaction must be built to first verify the VAA against the core bridge and subsequently call your program with the returned verified VAA.
